pkg/proxy/endpoints_test.go

```diff
@@ -1792,16 +1795,22 @@ func compareEndpointsMapsStr(t *testing.T, newMap EndpointsMap, expected map[Ser
 					t.Errorf("Failed to cast endpointsInfo")
 					continue
 				}
-				if *newEp != *(expected[x][i]) {
+				if !endpointEqual(newEp, expected[x][i]) {
 					t.Errorf("expected new[%v][%d] to be %v, got %v (IsLocal expected %v, got %v)", x, i, expected[x][i], newEp, expected[x][i].IsLocal, newEp.IsLocal)
 				}
 			}
 		}
 	}
 }
 
-func newTestEp(ep string) *BaseEndpointInfo {
-	return &BaseEndpointInfo{Endpoint: ep}
+func newTestEp(ep, host string) *BaseEndpointInfo {
+	endpointInfo := &BaseEndpointInfo{Endpoint: ep}
+	if host != "" {
+		endpointInfo.Topology = map[string]string{
+			"kubernetes.io/hostname": host,
```

Comment:
Because `Topology` need to be embedded into `Endpoint`,  and tests use `reflect.DeepEqual` to compare `Endpoint`, but expectedChange does not contains `Topology`, which computed change will contain host topology, so add a new parameter to this helper function to make expectedChange can contains host topology.